### PR TITLE
Added CORDIC algorithm

### PR DESCRIPTION
**Pull request by zivdar001matin**
### Describe your change:
I have added this algorithm so that it can help others to use it and get to know how simply CORDIC works.
Fixes: #7490 

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: other/cordic.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+SHIFT = 16
+SHIFT_BASE = 65536
+
+# (1/scale_factor) which scale factor is Π(cos(arctan(2^-i)))
+SCALE_CONSTANT = 1.64676
+
+"""
+The elementary angles for the lookup table. They correspond to
+arctan(2^-i) and then shifted left by 8 bits in an integer format
+"""
```
Comment (CaedenPH) [inline other/cordic.py:14]:
```suggestion
#  The elementary angles for the lookup table. They correspond to
#  arctan(2^-i) and then shifted left by 8 bits in an integer format
```

File: other/cordic.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        result = {"x": x, "y": y, "z": z}
+        return result
+
+    @staticmethod
+    def fixed_to_float(fixed: int) -> float:
+        return fixed / SHIFT_BASE
+
+    @staticmethod
+    def float_to_fixed(flt: float) -> int:
+        return (int)(flt * SHIFT_BASE)
```
Comment (CaedenPH) [inline other/cordic.py:164]:
```suggestion
        return int(flt * SHIFT_BASE)
```

**Review by CaedenPH — COMMENTED** [review]
Small changes, but otherwise looks awesome

**Comment by CaedenPH** [comment]
@cclauss Can we get your review please

**Comment by tianyizheng02** [comment]
@zivdar001matin you should also put `Fixes: #{$ISSUE_NO}` in your comment rather than just `#{$ISSUE_NO}` so that GitHub automatically closes the issue when this PR is merged

**Comment by zivdar001matin** [comment]
> @cclauss Can we get your review please

Sorry, I'm new to contributing and a little bit I'm confused. Could you please explain to me what you mean?

**Comment by zivdar001matin** [comment]
> @zivdar001matin you should also put `Fixes: #{$ISSUE_NO}` in your comment rather than just `#{$ISSUE_NO}` so that GitHub automatically closes the issue when this PR is merged

Thanks for your edit suggestion.

**Comment by tianyizheng02** [comment]
> > @zivdar001matin you should also put `Fixes: #{$ISSUE_NO}` in your comment rather than just `#{$ISSUE_NO}` so that GitHub automatically closes the issue when this PR is merged
> 
> Thanks for your edit suggestion.

It should not have backticks around them, otherwise GitHub won't recognize it. It should just look like this:
> Fixes: #7490

If it worked, then you should see something like the following on the right panel:
screenshot

**Comment by zivdar001matin** [comment]
> > > @zivdar001matin you should also put `Fixes: #{$ISSUE_NO}` in your comment rather than just `#{$ISSUE_NO}` so that GitHub automatically closes the issue when this PR is merged
> > 
> > 
> > Thanks for your edit suggestion.
> 
> It should not have backticks around them, otherwise GitHub won't recognize it. It should just look like this:
> 
> > Fixes: #7490
> 
> If it worked, then you should see something like the following on the right panel: screenshot

Oops x)))
Thank you a lot for explaining.

**Comment by zivdar001matin** [comment]
@cclauss Hi, I want this PR for the hacktoberfest event (Only 1 day left). Therefore, I’ll be thankful if we can get your review sooner.
Thanks in advance.

File: other/cordic.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+
+class CordicMode(Enum):
+    ROTATION = 0
+    VECTOR = 1
+
+
+class Cordic:
+    @staticmethod
+    def cos_cordic(theta: int) -> float:
```
Comment (tianyizheng02) [inline other/cordic.py:47]: Could we make numerical variables `float` or `int | float` rather than just `int`?

File: other/cordic.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        >>> round(Cordic.sin_cordic(30), 4)
+        0.5
+        """
+        return (
+            Cordic.fixed_to_float(Cordic.cordic(1, 0, theta, CordicMode.ROTATION)["y"])
+            / SCALE_CONSTANT
+        )
+
+    @staticmethod
+    def arctan_div_cordic(x: int, y: int) -> float:
```
Comment (tianyizheng02) [inline other/cordic.py:79]: For functions that pass in both the x- and y-coords of a vector, why not pass in the vector as a tuple?

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
1) Some refactoring suggestions (see comments)
2) This algorithm belongs more in the `maths/` directory than the `other/` directory since this is a numerical analysis algorithm
3) Add doctests to all functions

File: other/cordic.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    229,
+    115,
+    57,
+    29,
+    14,
+    7,
+    4,
+    2,
+    1,
+]
```
Comment (cclauss) [inline other/cordic.py:37]: WIll this work?  Tuples take less memory than lists and they are a signal to the reader that the contents will not change at runtime.

```suggestion
ELEM_ANGLE = (
    2949120,
    1740967,
    919789,
    466945,
    234379,
    117304,
    58666,
    29335,
    14668,
    7334,
    3667,
    1833,
    917,
    458,
    229,
    115,
    57,
    29,
    14,
    7,
    4,
    2,
    1,
)
```

File: other/cordic.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+
+class CordicMode(Enum):
+    ROTATION = 0
+    VECTOR = 1
+
+
+class Cordic:
```
[the remark this answers is not in the document]
Comment (cclauss) [reply other/cordic.py:45]: I am not a fan of this request.  I would rather stick with https://beta.ruff.rs/docs/rules/#pep8-naming-n

File: other/cordic.py (hunk 3 above)
Comment (cclauss) [reply other/cordic.py:47]: If so then some of our test need to use float input.

**Pull request closed without merging**